### DISCUSSION TITLE
Player command failed: Premium required

### DISCUSSION THREAD

**Discussion by mikebgrep** [spotify-web-api-java/spotify-web-api-java#332] [Q&A] [answered]
`se.michaelthelin.spotify.exceptions.detailed.ForbiddenException: Player command failed: Premium required`
I am trying to play a track from the player but receive this as response. Any hint why this is happened? 

This is how I authorize.

```   
public static void authorizationCode() throws InterruptedException {
        authorizationCodeUri();
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter code: ");
        String code = scanner.nextLine();
        Thread.sleep(20000);

        AuthorizationCodeRequest authorizationCodeRequest = spotifyApi.authorizationCode(code)
                .build();

        try {
            final AuthorizationCodeCredentials authorizationCodeCredentials = authorizationCodeRequest.execute();
            // Set access and refresh token for further "spotifyApi" object usage
            spotifyApi.setAccessToken(authorizationCodeCredentials.getAccessToken());
            spotifyApi.setRefreshToken(authorizationCodeCredentials.getRefreshToken());

            System.out.println("Expires in: " + authorizationCodeCredentials.getExpiresIn());
        } catch (IOException | SpotifyWebApiException | ParseException e) {
            System.out.println("Error: " + e.getMessage());
        }
    
```

**Comment by dargmuesli** [comment, marked as answer]
`Player command failed: Premium required` is feedback from Spotify's API, which means you do not have the necessary permissions. Please research this issue using [Spotify's documentation](https://developer.spotify.com/).

**Reply by mikebgrep** [reply]
Does this future need Premium account ? I give all permission to the app, and again I can't play the track or the album. If not, can you provide an example how to use it.

**Reply by dargmuesli** [reply]
See the README under https://github.com/spotify-web-api-java/spotify-web-api-java#authorization or head over to StackOverflow to find help on general authorization issues. This repository is meant to provide support for questions related to this repository's source code, not to the inner workings of Spotify's API.
